lib/sles4sap.pm

```diff
@@ -645,6 +646,37 @@ sub check_replication_state {
     die 'Timed out waiting for HANA System Replication to turn Active' unless ($time_to_wait > 0);
 }
 
+=head2 check_hanasr_attr
+
+ $self->check_hanasr_attr();
+
+Runs B<SAPHanaSR-showAttr> and checks in its output for up to a timeout
+specified in the named argument B<timeout> (defaults to 90 seconds) that
+the sync_state is B<SOK>. It also checks that no B<SFAIL> sync_status is
+present in the output. Finishes by printing the full output of
+B<SAPHanaSR-showAttr>. This method will only fail if B<SAPHanaSR-showAttr>
+returns a non-zero return value.
+
+=cut
+
+sub check_hanasr_attr {
+    my ($self, %args) = @_;
+    my $looptime = $args{timeout} // 90;
+    my $out;
+
+    while ($out = script_output 'SAPHanaSR-showAttr') {
+        last if ($out =~ /SOK/ && $out !~ /SFAIL/);
+        sleep bmwqemu::scale_timeout(3);
```

Comment:
`$looptime` is the global timeout, so is not better to do a `sleep 3;` here and modify L664 like this: `my $looptime = bmwqemu::scale_timeout($args{timeout} // 90);`? Otherwise for any value of `TIMEOUT_SCALE` the global timeout will still be 90s, no? I saw that `$looptime` will be decreased by 3 in all cases, but I found this way less understandable ;-)

Reply:
My idea was to wait at most "90 seconds" and those 90 seconds could be scaled, while at the same time limiting the number of cycles in the loop. So with 90/3 there are at most 30 cycles, but I scale the timeout in the `sleep`. You're right that this is a bit confusing. I'll move the scale timeout to L664 and pay no attention to the number of cycles in the loop.